crates/holochain_state/src/query/tests/links_test.rs

```diff
@@ -811,17 +821,29 @@ async fn link_type_ranges() {
     for d in &mut td {
         d.add_link_given_scratch(&mut scratch);
     }
-    TestData::only_these_on_query(&td, &scratch, .., here!("all return on full range"));
+    TestData::only_these_on_query(
+        &td,
+        &scratch,
+        LinkTypeFilter::Dependencies(td.iter().map(|d| d.zome_id).collect()),
+        here!("all return on full range"),
+    );
     TestData::only_these_on_query(
         &td[0..=0],
         &scratch,
-        LinkTypeRange::from(LinkType(0)..=LinkType(0)),
+        LinkTypeFilter::single_type(0.into(), 0.into()),
         here!("only single on single range"),
     );
     TestData::only_these_on_query(
         &td[4..=9],
         &scratch,
-        LinkTypeRange::from(LinkType(4)..=LinkType(9)),
+        LinkTypeFilter::Types(vec![
+            (4.into(), vec![4.into()]),
+            (5.into(), vec![5.into()]),
+            (6.into(), vec![6.into()]),
+            (7.into(), vec![7.into()]),
+            (8.into(), vec![8.into()]),
+            (9.into(), vec![9.into()]),
+        ]),
```

Comment:
I wonder if this points to us wanting a function for this variant, like:

```rust
fn types<A, B, Z, L>(a: A) -> LinkTypeFilter
where 
    A: Iterator<Item = (Z, B)>,
    B: Iterator<Item = L>,
    Z: Into<ZomeId>,
    L: Into<LinkType>
```

so then we can just say

```rust
LinkTypeFilter::types([(4, [4]), (5, [5]), .....])
```

Reply:
Ok I wasn't sure if zome devs would ever see this, it's ok if it's just for tests